crates/ibc/src/clients/ics07_tendermint/error.rs

```diff
@@ -105,7 +105,7 @@ pub enum Error {
     },
     /// trusted validators `{trusted_validator_set:?}`, does not hash to latest trusted validators. Expected: `{next_validators_hash}`, got: `{trusted_val_hash}`
     MisbehaviourTrustedValidatorHashMismatch {
-        trusted_validator_set: ValidatorSet,
+        trusted_validator_set: Box<ValidatorSet>,
```

Comment:
b7ff4fb4af33d10fdb6f6619190281d6171e2452

Reply:
Actually, I meant we should use the `Vec<Validator>` instead of `Vec<ValidatorSet>` because the hash is calculated using the `Vec<Validators>` :)  -> 
```diff
diff --git a/crates/ibc/src/clients/ics07_tendermint/client_state.rs b/crates/ibc/src/clients/ics07_tendermint/client_state.rs
index d1f47c0a..f6c492ee 100644
--- a/crates/ibc/src/clients/ics07_tendermint/client_state.rs
+++ b/crates/ibc/src/clients/ics07_tendermint/client_state.rs
@@ -282,7 +282,7 @@ impl ClientState {
 
         if trusted_consensus_state.next_validators_hash != trusted_val_hash {
             return Err(Error::MisbehaviourTrustedValidatorHashMismatch {
-                trusted_validator_set: Box::new(header.trusted_validator_set.clone()),
+                trusted_validator_set: header.trusted_validator_set.validators().clone(),
                 next_validators_hash: trusted_consensus_state.next_validators_hash,
                 trusted_val_hash,
             }
diff --git a/crates/ibc/src/clients/ics07_tendermint/error.rs b/crates/ibc/src/clients/ics07_tendermint/error.rs
index 9403eaed..38b0d2cb 100644
--- a/crates/ibc/src/clients/ics07_tendermint/error.rs
+++ b/crates/ibc/src/clients/ics07_tendermint/error.rs
@@ -12,11 +12,11 @@ use tendermint::account::Id;
 use tendermint::{Error as TendermintError, Hash};
 use tendermint_light_client_verifier::errors::VerificationErrorDetail as LightClientErrorDetail;
 use tendermint_light_client_verifier::operations::VotingPowerTally;
-use tendermint_light_client_verifier::types::ValidatorSet;
+use tendermint_light_client_verifier::types::Validator;
 use tendermint_light_client_verifier::Verdict;
 
 #[derive(Debug, Display)]
-pub enum Error {
+pub enum Errorw {
     /// chain-id is (`{chain_id}`) is too long, got: `{len}`, max allowed: `{max_len}`
     ChainIdTooLong {
         chain_id: ChainId,
@@ -105,7 +105,7 @@ pub enum Error {
     },
     /// trusted validators `{trusted_validator_set:?}`, does not hash to latest trusted validators. Expected: `{next_validators_hash}`, got: `{trusted_val_hash}`
     MisbehaviourTrustedValidatorHashMismatch {
-        trusted_validator_set: Box<ValidatorSet>,
+        trusted_validator_set: Vec<Validator>,
         next_validators_hash: Hash,
         trusted_val_hash: Hash,
     },
```
WDYT?

Reply:
Also, can we add another (breaking-change) changelog entry for this change in the public error type? (i.e. `.changelog/unreleased/breaking-changes`)

Reply:
oh, my bad. Fixed